[RTL] Set direction of HTML based on selected language on NS index page

watermandrewc opened the pull request:
### Description

On the program index page in North Star, render the HTML right-to-left for RTL languages. In some cases, styles were added in order for the RTL to render correctly (such as for checkboxes, which were still being rendered to the left of the text label before the additional CSS was added). 

These changes are not behind an additional feature flag since none of our instances support RTL languages yet.

Follow-up PRs will add tests and support for other pages (e.g. program overview, application edit page, etc.)

#6142

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [ ] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [ ] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).
- [x] Ensured PII wasn't added to any new logs, unless it was guarded by `isDevOrStaging`
- [x] Noted in the PR description which, if any, code in this PR was generated by AI.

#### User visible changes

- [ ] Followed steps to [internationalize](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#application-strings) any new strings
  - [ ] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [ ] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Made equivalent changes in Thymeleaf for applicant-facing features and changes (or created an issue in the [Northstar epic](https://github.com/orgs/civiform/projects/1/views/94) to track that it's missing from Thymeleaf)
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [ ] Manually evaluated tab order
- [ ] Manually tested with a screen reader if the feature is applicant-facing. See [screen reader testing](https://github.com/civiform/civiform/wiki/Testing#screen-reader-testing)

### Instructions for manual testing

1. Ensure the North Star UI feature flag is enabled.
2. Add a RTL language to the list of supported languages in the `application.conf` (e.g. "ar" for Arabic) and run the changes locally.
3. Select the RTL language from the "Select Language" dropdown.
4. Verify that the page is correctly rendered from right to left.

rockycodes reviewed (commented):
this looks great! i have some high level questions, so just commenting for now.

- is it possible to add browser tests for this? even one snapshot test of the homepage would be great.
- the styling is so simple. is it the `dir='rtl'` in the thymeleaf template that is handling most of it?

watermandrewc commented:
> this looks great! i have some high level questions, so just commenting for now.
> 
> * is it possible to add browser tests for this? even one snapshot test of the homepage would be great.

Added browser tests! They helped catch a few padding issues which are now fixed.

> * the styling is so simple. is it the `dir='rtl'` in the thymeleaf template that is handling most of it?

Yep! `dir='rtl'` handles most of the rendering; for anything else that still had any styling/padding issues, I updated the CSS.

> * re: `none of our instances support RTL languages yet`. is there any possibility of a gov enabled a RTL language before this work is done?

Technically it could be possible if they updated the supported languages, but it would still be rendered left to right. Would you prefer if this is flag-gated to prevent the RTL rendering?

rockycodes commented on browser-test/image_snapshots/northstar_applicant_application_index_test/applicant-homepage-right-to-left-desktop.png:
looks so good!!

rockycodes reviewed (approved):
Looks great! Would you add the "under-development" label before merging? I think it's fine to merge this without a feature flag since technically govts could add an RTL language right now, it would just render LTR as you said. But I don't want a government seeing our next release and thinking we fully support RTL languages now, so better to keep it in the "under development" release section until it's completed.
